lib/ecto/query/builder.ex

```diff
@@ -281,7 +281,7 @@ defmodule Ecto.Query.Builder do
     rtype = quoted_type(left, vars)
 
     {left,  params_acc} = escape(left, ltype, params_acc, vars, env)
-    {right, params_acc} = escape(right, rtype, params_acc, vars, env)
+    {right, params_acc} = escape_subquery(right, rtype, params_acc, vars, env)
```

Comment:
I don't think you need this change. :)

Reply:
This is needed to support the case where we don't want to use a quantifier. For example:

```
avg_rating = from(p in Post, select: avg(p.rating))
from(p in Post, where: p.rating > subquery(avg_rating))
```

Without this we get
```
** (Ecto.Query.CompileError) `subquery(query)` is not a valid query expression.
```

Reply:
Oh, you also want to support row constructors. I see. :) I would skip it for now since we don't support the `ROW` operation built-in. We can always add it later in the future though if someone has a proper use case. :)

Reply:
I see what you're referring too, yes row constructors are related to this work. I didn't know you could do this type of comparison with multiple fields using row constructors https://dev.mysql.com/doc/refman/8.0/en/row-subqueries.html

For now I'm just dealing with the [scalar case](https://dev.mysql.com/doc/refman/8.0/en/comparisons-using-subqueries.html) where the subquery must return a single column and be compared to a scalar expression. I use this a lot for use cases like finding "above average" rows, rows that are in certain percentiles and stuff like that.

By the way, I also tested all these queries locally with an old PHX project and they generate valid SQL without any adapter changes =).

Reply:
So I don't want to work on supporting ROW constructors yet, I've never seen this used constructively so it's best to just use a fragment for that kind of thing, but the scalar comparison is not too difficult to support with the groundwork that's already been laid out.

Reply:
If those queries have to return a single column and a single row, then it is equivalent `foo >= all(sub)` anyway, no?

Reply:
Good point. I guess databases support both syntaxes for backwards compatibility, we don't necessarily need to. Since it's one less thing to maintain and worry about I will drop the `>= subquery` syntax.